crates/css_ast/src/types/transform_list.rs

```diff
@@ -1,32 +1,75 @@
-#![allow(warnings)]
-use css_lexer::{Cursor, SourceOffset};
-use css_parse::{CursorSink, Parse, Parser, Peek, Result as ParserResult, T, ToCursors};
+use bumpalo::collections::Vec;
+use css_lexer::Cursor;
+use css_parse::{CursorSink, Parse, Parser, Peek, Result as ParserResult, ToCursors};
+
+use crate::TransformFunction;
 
 // https://drafts.csswg.org/css-transforms-1/#typedef-transform-list
 // <transform-list> = <transform-function>+
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize), serde())]
-pub struct TransformList;
+pub struct TransformList<'a>(Vec<'a, TransformFunction>);
 
-impl<'a> Peek<'a> for TransformList {
+impl<'a> Peek<'a> for TransformList<'a> {
 	fn peek(p: &Parser<'a>, c: Cursor) -> bool {
-		todo!();
+		TransformFunction::peek(p, c)
 	}
 }
 
-impl<'a> Parse<'a> for TransformList {
+impl<'a> Parse<'a> for TransformList<'a> {
 	fn parse(p: &mut Parser<'a>) -> ParserResult<Self> {
-		todo!();
+		let mut list = Vec::new_in(p.bump());
+		while let Some(transform) = p.parse_if_peek::<TransformFunction>()? {
+			list.push(transform);
+		}
+		Ok(TransformList(list))
 	}
 }
 
-impl<'a> ToCursors for TransformList {
+impl<'a> ToCursors for TransformList<'a> {
 	fn to_cursors(&self, s: &mut impl CursorSink) {
-		todo!();
+		for transform in &self.0 {
+			ToCursors::to_cursors(transform, s);
+		}
 	}
 }
 
 #[cfg(test)]
 mod tests {
 	use super::*;
+	use css_parse::{assert_parse, assert_parse_error};
+
+	#[test]
+	fn size_test() {
+		assert_eq!(std::mem::size_of::<TransformList>(), 32);
+	}
+
+	#[test]
+	fn test_writes() {
+		assert_parse!(TransformList, "matrix(1,0,0,1,0,0)");
+		assert_parse!(TransformList, "translate(1rem)");
+		assert_parse!(TransformList, "translateX(1rem)");
+		assert_parse!(TransformList, "translateY(1rem)");
+		assert_parse!(TransformList, "scale(2)");
+		assert_parse!(TransformList, "scale(1,2)");
+		assert_parse!(TransformList, "scaleX(2)");
+		assert_parse!(TransformList, "scaleY(2)");
+		assert_parse!(TransformList, "rotate(45deg)");
+		assert_parse!(TransformList, "skew(1deg,2deg)");
+		assert_parse!(TransformList, "skewX(1deg)");
+		assert_parse!(TransformList, "skewY(1deg)");
+		assert_parse!(TransformList, "rotate(180deg)scale(2,3)");
+		assert_parse!(TransformList, "skewX(10deg)skewY(20deg)rotate(45deg)");
+		assert_parse!(TransformList, "scale(1.5)rotate(90deg)skew(15deg,30deg)");
+		assert_parse!(
+			TransformList,
+			"matrix(1,0,0,1,0,0)translate(1rem)translateX(1rem)translateY(1rem)scale(2)scale(1,2)scaleX(2)scaleY(2)rotate(45deg)skew(1deg,2deg)skewX(1deg)skewY(1deg)"
```

Comment:
Just a test to ensure everything can be chained. But question @keithamus, are whitespaces lost in tokens? Be kinda nice to be able to:

```diff
- assert_parse!(TransformList, "rotate(180deg)scale(2,3)");
+ assert_parse!(TransformList, "rotate(180deg) scale(2,3)");
```

Tests fail coz its missing a space.